src/coreclr/scripts/superpmi_benchmarks.py

```diff
@@ -206,6 +206,38 @@ def build_and_run(coreclr_args, output_mch_name):
          "--framework", "net8.0", "--no-restore", "/p:NuGetPackageRoot=" + artifacts_packages_directory,
          "-o", artifacts_directory], _exit_on_fail=True)
 
+    # This is specifically for PowerShell.Benchmarks.
+    # Because we are using '--corerun' when running the benchmarks, it is not able to resolve the PowerShell dependent assemblies.
+    # To make this work, we create a directory called 'core_root' in the directory where the benchmarks_dll lives.
+    # Then we copy the relevant PowerShell dependencies to 'core_root'.
+    # Then we copy the contents of the original 'core_root' to the benchmarks' 'core_root', potentially overwriting any older core dlls.
+    # Then we just use the 'corerun.exe' from the benchmarks' 'core_root'.
+    # We make a copy of 'core_root' as to avoid any potential file-system issues.
+    if benchmarks_dll.endswith("PowerShell.Benchmarks.dll"):
+        benchmarks_dll_dir = os.path.dirname(benchmarks_dll)
+
+        # Create benchmarks' 'core_root'.
+        benchmarks_dll_core_root = os.path.join(benchmarks_dll_dir, "core_root")
+        if not os.path.exists(benchmarks_dll_core_root):
+            os.makedirs(benchmarks_dll_core_root)
+
+        # Begin copy PowerShell dependencies.
+        copy_directory(os.path.join(benchmarks_dll_dir, "runtimes/win/lib/net7.0/"), benchmarks_dll_core_root, verbose_copy=True)
```

Comment:
Should you always copy the `win` dependencies even for Linux/Mac?

It seems potentially fragile to hard-code `net7.0`, but maybe we won't change the PS benchmark code? (Or rarely change it)

Reply:
It it ok to to a tree copy? i.e., you're copying sub-directories that will remain sub-directories (e.g., I notice there's a "Modules" sub-directory).

Reply:
Will look into this more.

I will admit that hard-coding net7.0 is fragile, but I know no other way, other than changing how we run these benchmarks using a custom 'dotnet.exe' that is backed by the Checked build where it can resolve assemblies successfully.